presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/MapStatistics.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata.statistics;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.orc.metadata.statistics.StatisticsHasher.Hashable;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class MapStatistics
+        implements Hashable
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(MapStatistics.class).instanceSize();
+    private final List<MapStatisticsEntry> entries;
```

Comment:
Using a list here means that we make the stats equals/hashCode dependent on the order of the keys. I'm not sure that this is a right solution. 

The only place that would use equals/hashCode is the writer validation, the rest of the places don't care about the equality or the order. That's why I'm hesitant to use a map for this field, because it will impact performance and memory consumption. 

If needed I can switch equals and hashCode to use a hash map for entries, but keep entries as a list field. Opinion?

Reply:
My bad, did not see your comment and commented the same thing. I am fine with List.